newIDE/app/src/EventsSheet/index.js

```diff
@@ -267,6 +271,18 @@ export class EventsSheetComponentWithoutHandle extends React.Component<
 
   componentDidMount() {
     this.setState({ allEventsMetadata: enumerateEventsMetadata() });
+    if (this.props.leaderboardsManager) {
+      this.props.leaderboardsManager
```

Comment:
Is there any chance we could do this inside the inline renderers? 
Or at least, have the inline renderers do that once (with a useEffect) only if the leaderboards are not loaded?

I'm not sure if it would work, I'm just trying to remove as much logic as possible from this poor EventsSheet

Reply:
I tried fetching in inlineRenderer, with a fetch only if leaderboards is null, but it still called the API once for every occurence of the field in the EventsSheet the first time it renders (My guess is that the first time, all the useEffect are called at the same time and leaderboards are still null at the time). 😕

Reply:
This will be treated in the next PR (here : https://github.com/4ian/GDevelop/pull/3779) that will move the listLeaderboards the API call from EventsSheet to the inlineField/ParameterField and ensure that only one call is made at once by adding a variable in the Provider component.
(More specific style typing too)

Reply:
Sounds good 👍